week-3-data-join.html

```diff
@@ -3,281 +3,222 @@
   var tbody = d3.select("tbody");
+    function updateTable(){
+        var data = voting.slice();
 
-
-  function updateTable(){
-
-    // Erase any existing html code within tbody by replacing it with a blank string "".
     tbody.html("");
-
-    // Binding data - this will be a very common task in our use of D3. Start by making a selectAll of <tr> elements that do not yet exist (they will soon!).
-    // Then we bind our data (schools) with .data and .enter()
-    // Lastly, we append the <tr> elements, which are become our current selection because of the prior selectAll("tr")
+    
     var rows = tbody.selectAll("tr")
-      .data(schools)
+      .data(voting)
       .enter()
       .append("tr");
 
-    // If you run, console.log(rows), you should now see an array of <tr> elements printed out to the console, each of which should contain a __data__ property containing one object from our JSON array.
-    // console.log(rows);
+    rows.append("td")
+      .html(function(d) { return d.state });
+
+    rows.append("td")
+      .html(function(d) { return d.region });
 
-    // Take the rows variable and append a <td> elements containing d.name - the name of each schools for each bound data element.
     rows.append("td")
-      .html(function(d) { return d.name });
+      .html(function(d) { return d.repVotes });
 
-    // Same as above, except for level of the school.
     rows.append("td")
-      .html(function(d) { return d.level_text });
+      .html(function(d) { return d.totalVotes });
 
-    // Same as above, except for the number of students enrolled in the school.
     rows.append("td")
-      .html(function(d) { return d.enrollment + ' students' });
+      .html(function(d) { return d.percentage });
 
-    // Same as above, calculating a percent of the space allowed that the div should fill:
     rows.append("td")
       .append("div")
       .classed("bar", true)
-      .style("width", function(d) { return ((d.enrollment / MAX_ENROLLMENT) * 100) + '%' })
-
+      .style("width", function (d) { return ((d.repVotes / d.totalVotes) * 100) + '%' })
   };
 
   updateTable();
 
+// d3 sorting
 
-  // We can use d3.on to trigger events - note that these just change the class of the buttons when clicked, they do not trigger any other events. Still, you should be able to click on the buttons and see them change (become darker as through depressed).
-  
   d3.select("#sort-asc")
     .on("click", function(){
       d3.select('#sort-desc').classed('active', false);
       d3.select('#sort-asc').classed('active', true);
+      tbody.selectAll("tr").sort(function (a,b){
+      return d3.ascending(a.perc, b.perc);  });
     });
 
   d3.select("#sort-desc")
-    .on("click", function(){
+    .on("click", function (){ 
       d3.select('#sort-asc').classed('active', false);
       d3.select('#sort-desc').classed('active', true);
+      tbody.selectAll("tr").sort(function (a,b){
+      return d3.descending(a.perc, b.perc);  });
     });
 
```

Comment:
So the above code works for sorting, though it wasn't the way we originally imagined. 

We were trying to get you to grab the code from last week (week 2) that transforms the data based on options within the updateTable() function. The buttons themselves would then simply change the options and rerun updateTable(). That's what Chris was trying to get to in his email on Wednesday.

The reason this is important is that sort and filter work differently. Sort is going to rearrange the DOM elements all on its own. Filter isn't going to do this however, it just filters the selection that the data is bound to. This is why we need to filter, then run updateTable again - clearing and repopulating all the table elements.